src/util/exception_utils.h

```diff
@@ -38,4 +43,31 @@ class invalid_user_input_exceptiont
   }
 };
 
+
+class equation_conversion_exceptiont : public std::runtime_error
```

Comment:
If the intent of moving this exception class into this file was to make it part of the new exception class hierachy, then this class should be inheriting from something other than std::runtime_error...... I whole heartedly agree that this class *should* move into the new hierachy, but it *shouldn't* be in this util directory. It should stay in it's original file but just inherit from a suitable class in the new hierachy.

Reply:
There's no hierarchy yet, and thinking about how to model one I think it's trickier than it looks to just design one on the spot. I will just revert this commit and note it down as one of the improvements I want to take a look at further down the line.